From a5bac0121d9e3e5b6f011861ba1a95365528619d Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 3 Mar 2015 11:10:48 +0800
Subject: [PATCH] mtd: m25p80: skip set_quad_mode for Zynq QSPI flash.

It is necessary for xilinx-zynq QSPI flash to skip set_quad_mode
since its jedec_id is the same with ST flash jedec_id rather than
Spansion.

Signed-off-by:Zumeng Chen <zumeng.chen@windriver.com>

diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index d76859f2d664..fe5b870034fe 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -1383,7 +1383,6 @@ static int m25p_probe(struct spi_device *spi)
 	unsigned			i;
 	struct mtd_part_parser_data	ppdata;
 	struct device_node *np = spi->dev.of_node;
-	int ret;
 
 	/* Platform data helps sort out which chip type we have, as
 	 * well as how this board partitions it.  If we don't have
@@ -1570,11 +1569,14 @@ static int m25p_probe(struct spi_device *spi)
 
 	/* Quad-read mode takes precedence over fast/normal */
 	if (spi->mode & SPI_RX_QUAD && info->flags & M25P80_QUAD_READ) {
+#ifndef CONFIG_SPI_ZYNQ_QSPI
+		int ret;
 		ret = set_quad_mode(flash, info->jedec_id);
 		if (ret) {
 			dev_err(&flash->spi->dev, "quad mode not supported\n");
 			return ret;
 		}
+#endif
 		flash->flash_read = M25P80_QUAD;
 	}
 
-- 
2.1.0

